selendroid-server/src/main/java/io/selendroid/server/model/DefaultAppiumUiAutomatorDriver.java

```diff
@@ -0,0 +1,305 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.selendroid.server.model;
+
+import android.support.test.uiautomator.UiDevice;
+import android.test.InstrumentationTestCase;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.UUID;
+
+import io.selendroid.server.ServerInstrumentation;
+import io.selendroid.server.android.InstrumentedKeySender;
+import io.selendroid.server.android.KeySender;
+import io.selendroid.server.android.internal.Dimension;
+import io.selendroid.server.common.exceptions.NoSuchMethodImplementationException;
+import io.selendroid.server.common.exceptions.SelendroidException;
+import io.selendroid.server.util.Preconditions;
+import io.selendroid.server.util.SelendroidLogger;
+
+
+public class DefaultAppiumUiAutomatorDriver implements SelendroidDriver {
+
+    public static final String BROWSER_NAME = "browserName";
+    public static final String PLATFORM = "platform";
+    public static final String SUPPORTS_JAVASCRIPT = "javascriptEnabled";
+    public static final String TAKES_SCREENSHOT = "takesScreenshot";
+    public static final String SUPPORTS_ALERTS = "handlesAlerts";
+    public static final String ROTATABLE = "rotatable";
+    public static final String ACCEPT_SSL_CERTS = "acceptSslCerts";
+    public static final String SUPPORTS_NETWORK_CONNECTION = "networkConnectionEnabled";
+    public static final String AUTOMATION_NAME = "automationName";
+    public static final String PLATFORM_NAME = "platformName";
+    public static final String PLATFORM_VERSION = "platformVersion";
+    public static final String SERVER_VERSION = "version";
+    public static final String METHOD_NOT_IMPLMENTED_EXCEPTION = "Method has no Implementation";
+    private ServerInstrumentation serverInstrumentation = null;
+    private KeySender keySender = null;
+    private Session session = null;
+
+
+    private UiDevice mDevice;
+    private SelendroidWebDriver selendroidWebDriver = null;
+
+    public DefaultAppiumUiAutomatorDriver(ServerInstrumentation serverInstrumentation) {
+        this.keySender = new InstrumentedKeySender(serverInstrumentation);
+        this.serverInstrumentation = serverInstrumentation;
+        // An exception is thrown right here
+        // mDevice = UiDevice.getInstance(serverInstrumentation);
+    }
+
+    @Override
+    public AndroidElement findElement(By by) {
+        mDevice = UiDevice.getInstance(serverInstrumentation);
```

Comment:
> Was this ever solved?

Nope. I was hoping one of the selendroid developers would have time to investigate but everyone is busy. We've had to fork selendroid and work on the code there.